lib/rubocop/cop/lint/to_json.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks to make sure `#to_json` includes an optional hash
+      # argument.  When overriding `#to_json`, callers may invoke JSON
+      # generation via `JSON.generate(your_obj)`.  Since `JSON#generate` allows
+      # for optional arguments, your method does too.
+      #
+      # @example
+      #   # bad
+      #   def to_json
+      #   end
+      #
+      #   # good
+      #   def to_json(opts = {})
+      #   end
+      #
+      class ToJSON < Cop
+        MSG = <<-M_JSON.freeze
+        `#to_json` requires an optional hash argument to be parsable via JSON.generate(obj).
+        M_JSON
+
+        def on_def(node)
+          return if node.method_name != :to_json
+          return unless node.arguments.empty?
+
+          add_offense(node)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.insert_after(node.loc.name, '(_opts = {})')
```

Comment:
Think this will raise a RuboCop offense in `Style/OptionsHash`. We should probably look at adding a whitelist entry to that cop to account for this.

Reply:
@Drenmi thanks for calling this out - where exactly would I add the whitelist?

Reply:
@Drenmi can you point me in the right direction?

Reply:
@allcentury: The changes need to be made in `Style/OptionHash`. For an example of a method whitelist, check e.g. `Rails/DynamicFindBy`.

Reply:
@allcentury: The changes need to be made in `Style/OptionHash`. For an example of a method whitelist, check e.g. `Rails/DynamicFindBy`.

Reply:
@allcentury: The changes need to be made in `Style/OptionHash`. For an example of a method whitelist, check e.g. `Rails/DynamicFindBy`. 🙂

Reply:
@Drenmi thank you!  I think I've implemented your suggestion, please take a look when you have a sec